src/Dialog/Dialog.test.tsx

```
@@ -0,0 +1,85 @@
+import React from 'react';
+import { render, screen, act, fireEvent, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { Dialog, DialogProps } from './Dialog';
+import 'jest-styled-components';
+
+const ANIMATION_DURATION = 200;
+
+const TestDialog = (props: Partial<DialogProps>) => (
+    <Dialog
+        trigger={(openDialog) => <button onClick={openDialog}>Open Dialog</button>}
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)